pkg/clusterconditions/promql/promql.go

```diff
@@ -61,12 +68,22 @@ func NewPromQL(kubeClient kubernetes.Interface) *cache.Cache {
 // on the pod before DNS pods are available and before the service network is available.  The side effect is that
 // the CVO cannot resolve service DNS names.
 func (p *PromQL) Address(ctx context.Context) (string, error) {
-	svc, err := p.kubeClient.CoreV1().Services("openshift-monitoring").Get(ctx, "thanos-querier", metav1.GetOptions{})
+	svc, err := p.kubeClient.CoreV1().Services(p.QueryNamespace).Get(ctx, p.QueryService, metav1.GetOptions{})
 	if err != nil {
 		return "", err
 	}
-
-	return fmt.Sprintf("https://%s", net.JoinHostPort(svc.Spec.ClusterIP, "9091")), nil
+	var port string
+	for i := range svc.Spec.Ports {
+		if svc.Spec.Ports[i].Name == "web" {
+			port = strconv.Itoa(int(svc.Spec.Ports[i].Port))
+			break
+		}
+	}
+	scheme := "https"
+	if p.QueryNamespace == "openshift-observability-operator" && p.QueryService == "hypershift-monitoring-stack-prometheus" {
+		scheme = "http"
```

Comment:
:monocle_face: why don't we have TLS in hypershift?

Reply:
Or I am using a wrong configuration, or the service is not expected to be queried. Investigating...

Was initially going from the [comment](https://github.com/openshift/cluster-version-operator/pull/926#issuecomment-1633279250):

> Follow steps on https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-57234 for Observability Operator installation

> Follow steps on https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-57236 to create MonitoringStack CR to collect HyperShift hosted-control-plane metrics. The secret and the spec.prometheusConfig.remoteWrite field can be omitted.

Reply:
Http looks odd. @wking Do you know if we have SSL TLS auth for the monitoring stack?